controllers/metal3.io/baremetalhost_controller.go

```diff
@@ -1161,6 +1188,243 @@ func saveHostProvisioningSettings(host *metal3v1alpha1.BareMetalHost) (dirty boo
 	return
 }
 
+// Get the stored firmware settings and validate the Spec settings against the schema
+func (r *BareMetalHostReconciler) getHostFirmwareSettings(prov provisioner.Provisioner, info *reconcileInfo) (hfs *metal3v1alpha1.HostFirmwareSettings, err error) {
+
+	hfs = &metal3v1alpha1.HostFirmwareSettings{}
+	if r.Get(context.TODO(), info.request.NamespacedName, hfs) != nil {
+		// Could not get settings, don't return error as settings may not have been available at provisioner
+		info.log.Info("could not get hostFirmwareSettings", "namespacename", info.request.NamespacedName)
+		return nil, nil
+	}
+
+	if hfs.Status.Settings == nil {
+		// If settings cleared after pivot, recreate map and update settings
+		hfs.Status.Settings = make(metal3v1alpha1.SettingsMap)
+
+		currentSettings, _, err := prov.GetFirmwareSettings(false)
+		if err != nil {
+			info.log.Info("could not get firmware settings from provisioner")
+			return nil, nil
+		}
+		if err = r.updateHostFirmwareSettingsFromCurrent(currentSettings, hfs, nil); err != nil {
```

Comment:
nit: it's a bit weird that a method called `getSomething` also does an update.

Reply:
Yeah I agree, but we need to update the Status fields if they get removed after a pivot. i haven't found names that would express this better. Still thinking about it...